test/coreneuron/test_spikes.py

```diff
@@ -119,17 +118,16 @@ def run(mode):
     # CORENEURON run
     from neuron import coreneuron
 
-    with coreneuron(enable=True, gpu=enable_gpu, file_mode=file_mode, verbose=0):
+    with coreneuron(enable=True, gpu=enable_gpu, file_mode=file_mode):
         run_modes = [0] if file_mode else [0, 1, 2]
         for mode in run_modes:
             run(mode)
-        # Make sure that file mode also works with custom coreneuron.data_path
+        # Make sure that file mode also works with custom coreneuron.model_path
         if file_mode:
-            temp_coreneuron_data_folder = tempfile.TemporaryDirectory(
-                "coreneuron_input"
-            )  # auto removed
-            coreneuron.data_path = temp_coreneuron_data_folder.name
+            coreneuron.model_path = "coreneuron_input"
```

Comment:
In fact, this test tends to run coreneuron runs twice with `file_mode=1`. The first run writes the coreneuron data in the default folder `cornrn_data`, and the second run with the custom folder given by `model_path`. (NB, `data_path` is not a valid key).
As I see in local, the mpi enabled run very often encountered the race condition for the second run: when rank 0 is writing the `files.dat`, rank 1 already starts to read it. I've no clue why always the second. However, I see that situation only with the coreneuron quiet mode `verbose=0`. Without the quite mode, it seems we print the memeory usage, in this case `report_mem_usage("After mk_mech");` seem to help to hold the other ranks to wait for the data writing. 
I guess we need to check the process of the coreneuron data writing, maybe set a barrier at the end?